api/core/v1beta2/common_types.go

```diff
@@ -405,3 +408,49 @@ func (r *ContractVersionedObjectReference) GroupKind() schema.GroupKind {
 		Kind:  r.Kind,
 	}
 }
+
+// MachineTaint defines a taint equivalent to corev1.Taint, but additionally having a propagation field.
+type MachineTaint struct {
+	// key is the taint key to be applied to a node.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Key string `json:"key,omitempty"`
+
+	// value is the taint value corresponding to the taint key.
```

Comment:
This should be validated as a label value, please apply the correct regex or format util from CEL

Reply:
Sounds good. Same comment from my side as for key. Let's use regexp for now, we'll move to to the CEL func soon